Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/System/RenderPipelineDescriptor.h

```diff
@@ -24,6 +24,8 @@ namespace AZ
             AZ_TYPE_INFO(RenderPipelineDescriptor, "{B1A5CF41-AC8D-440E-A1E9-3544D7F3445B}");
             static void Reflect(AZ::ReflectContext* context);
 
+            static constexpr AZStd::string_view Extension = "RenderPipeline.azasset";
```

Comment:
Personally, I'm not a fan of the coercion from a statically embedded `const char*` string literal to a string view (I believe this is one of several `string_view` design flaws). This usage implies that users of the string view may depend on the presence of a `NUL` terminator, which is an implicit compatibility burden and future footgun (what happens when someone passes in a `string_view` which isn't nul-terminated?). Even though a `char*` isn't necessarily nul-terminated, this is at the very least a well-established convention and communicates an expectation to the users of the type. Prefer the very not fancy `const char*` literal (I'm aware this pattern is abused in other places in the codebase, it's something we should root out over time IMO)

Reply:
I just copied and pasted from elsewhere, but I can update this and the original.

Reply:
Oh wow, I found 161 hits of this pattern and only 13 of them are in Atom. I'll just update the couple ones I coped from.

Reply:
I have historically not been a fan of using string_view globally for all of the reasons you mentioned. However, it has worked beautifully and optimally in the few cases that I used it recently, comparing file extensions at the end of strings. Otherwise, I have avoided it because it requires additional checks and special handling everywhere it gets used as opposed to const AZStD::string&. I can see the benefits if you want the interface of a string without the allocations or to be able to treat different types of strings uniformly.

Reply:
At some point we might want to look into using AZStd::fixed_string for static consts, to get the best of both worlds.